src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -223,6 +220,9 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
     task = client.get(resource_group_name, registry_name, task_name)
     step = task.step
     if isinstance(step, DockerBuildStep):
+        if file.endswith(ALLOWED_TASK_FILE_TYPES):
```

Comment:
these checks seem superfluous since you've guaranteed step is a `DockerBuildStep` if it doesn't end with it above.